libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,34 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	FunctionDefinition const* function = *_unaryOperation.annotation().userDefinedFunction;
+	if (function)
+	{
+		solAssert(function->isFree());
+
+		FunctionType const* functionType = _unaryOperation.userDefinedFunctionType();
+		solAssert(functionType);
+		solAssert(functionType->parameterTypes().size() == 1);
+		solAssert(functionType->returnParameterTypes().size() == 1);
+		solAssert(functionType->kind() == FunctionType::Kind::Internal);
+
+		evmasm::AssemblyItem returnLabel = m_context.pushNewTag();
+		acceptAndConvert(_unaryOperation.subExpression(), *functionType->parameterTypes()[0]);
```

Comment:
```suggestion
		acceptAndConvert(_unaryOperation.subExpression(), *functionType->parameterTypes()[0], true);
```

Reply:
Left as is. See https://github.com/ethereum/solidity/pull/13790#discussion_r1113626349.